lib/bolt/inventory/group.rb

```diff
@@ -239,6 +239,15 @@ def validate_group_input(input)
           raise ValidationError.new(msg, @name)
         end
 
+        if input.key?('nodes')
+          msg = <<~MSG
+                Found 'nodes' key in group #{@name}. This looks like a v1 inventory file, which is
+                no longer supported by Bolt. Migrate to a v2 inventory file automatically using
+                'bolt project migrate'.
+                MSG
+          raise ValidationError.new(msg, @name)
```

Comment:
`ValidationError` leads to some weird behavior here because it appends "for group at [<path to group>]" to the message. So I got this:

```
Found 'nodes' key in group all. This looks like a v1 inventory file, which is
no longer supported by Bolt. Migrate to a v2 inventory file automatically using
'bolt project migrate'.
 for group at ["all"]
```

That can be avoided by passing `nil` instead of `@name`. That's reasonable because this is a problem with the _whole inventory file itself_, not just this group. Or at least that's how I'll rationalize it to myself.

Also, this message has a trailing newline which winds up being printed. We should add a `.chomp` after `<<~MSG`.